### PR TITLE
Fixed incorrect depsdev getProject

### PR DESCRIPTION
**Pull request by nathannaveen**
# Description of the PR

This PR addresses the issue described in #1413. It is based on the changes from PR #1438.

## Reason for New PR:

I do not have permissions to push to the original PRs branch, so I am creating this new pull request to continue the work.

# PR Checklist

- [ ] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If GraphQL schema is changed, GraphQL client updates/additions have been made
- [ ] If OpenAPI spec is changed, `make generate` has been run
- [ ] If ent schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [ ] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Comment by pxp928** [comment]
@mdeicas can you take a look at this as you reviewed the original PR and opened the issue.

**Review by mihaimaruseac — COMMENTED** [review]
This LGTMs, but leaving second approval to @mdeicas

**Review by mdeicas — APPROVED** [review]
Thanks for fixing this up @nathannaveen. 

A few things I noticed while reviewing:
- The issue described in #1413 no longer seems to exist, at least for the example given there. The example call to `GetProject` now succeeds both with and without the suffix.
- I asked the deps.dev team about the above, and was pointed to an improvement they've made in the mapping from package version to project -- the `GetVersions` endpoint now explicitly returns the link. See this change https://github.com/google/deps.dev/commit/1729b6244a2d8e484397ca8a095613ee0a64b22f#diff-0d17ca7a12c7399a3118d28d271b2851746c3b2270652c90213f5225c189e41aR278 and the documentation for `relatedProjects` in https://docs.deps.dev/api/v3alpha/#getversion. `relatedProjects.projectKey` can now be used to reliably construct the arguments to `GetVersions`. 

We should pick up these additions to the API (I'll file an issue) in order to simplify these calls and for the additional information about the source of the project mapping (https://github.com/google/deps.dev/blob/c5a1406b04702d093b9186c9129d4331c93e1c92/api/v3alpha/api.proto#L963). Doing so would eliminate #1413 as a potential bug. 

I'm OK with merging these changes in the meantime (e.g. as a regression test against that bug), but I wonder if the removal of the .git suffix would become unnecessary and the `TestDepsCollector_collectAdditionalMetadata` test redundant once we update to the API additions. What do you think?

**Comment by nathannaveen** [comment]
@mdeicas, given the information that you have provided, I agree that the removal of the .git suffix would be unnecessary once the new additions to the API are implemented. Part of the issue could be to remove the removal of the `.git` suffix as well as the redundant test.

File: pkg/handler/collector/deps_dev/deps_dev_test.go [hunk 1]
```diff
@@ -478,3 +491,70 @@ func TestProjectKey(t *testing.T) {
...
+		wantLog      string
+	}{
+		{
+			testName:     "npm package with .git suffix",
+			pkgType:      "npm",
+			namespace:    ptrfrom.String("@webassemblyjs"),
+			name:         "wasm-parser",
+			version:      ptrfrom.String("1.11.6"),
+			pkgComponent: &PackageComponent{},
+			wantLog:      "",
```
[the remark this answers is not in the document]
Comment (nathannaveen) [reply pkg/handler/collector/deps_dev/deps_dev_test.go]: Thanks for the review! I realized the test you commented on was redundant. As the other test covered the same case.

**Pull request merged**